discovery/kubernetes/kubernetes.go

```diff
@@ -250,13 +254,22 @@ type Discovery struct {
 	namespaceDiscovery *NamespaceDiscovery
 	discoverers        []discovery.Discoverer
 	selectors          roleSelector
+	ownNamespace       string
 }
 
 func (d *Discovery) getNamespaces() []string {
-	namespaces := d.namespaceDiscovery.Names
+	namespaces := make([]string, len(d.namespaceDiscovery.Names))
 	if len(namespaces) == 0 {
 		namespaces = []string{apiv1.NamespaceAll}
 	}
+
+	copy(namespaces, d.namespaceDiscovery.Names)
+	for i, namespace := range namespaces {
+		if namespace == discoveryOwnNamespace {
+			namespaces[i] = d.ownNamespace
```

Comment:
We could also do the replacement when unmarshaling the config, I don't have a clear idea of what would be better in this case.